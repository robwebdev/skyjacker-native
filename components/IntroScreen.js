import * as Animatable from "react-native-animatable";

import {
  Alert,
  Image,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from "react-native";

import Colors from "../constants/Colors";
import React from "react";
import TextStyles from "../styles/Text";

const AniamtedTouchableOpacity = Animatable.createAnimatableComponent(
  TouchableOpacity
);

export default ({
  bodyText,
  image,
  titleText,
  buttonText,
  navigation,
  nextScreen
}) => {
  return (
    <Animatable.View
      animation="fadeInRightBig"
      duration={500}
      style={styles.container}
    >
      <Image source={image} style={styles.image} />
      <Animatable.View
        style={{ alignItems: "center" }}
        animation="pulse"
        delay={500}
      >
        <Text style={[styles.text, { color: Colors.YELLOW }]}>{titleText}</Text>
        <Text style={styles.text}>{bodyText[0]}</Text>
        <Text style={styles.text}>{bodyText[1]}</Text>
        <AniamtedTouchableOpacity
          animation="fadeIn"
          delay={1000}
          style={styles.button}
          onPress={toNextStep}
        >
          <Text style={[styles.text, { color: "black", textAlign: "center" }]}>
            {buttonText}
          </Text>
        </AniamtedTouchableOpacity>
      </Animatable.View>
    </Animatable.View>
  );

  function toNextStep() {
    if (nextScreen) {
      navigation.dispatch({
        key: nextScreen,
        type: "ReplaceCurrentScreen",
        routeName: nextScreen
      });
    } else {
      Alert.alert("Thatâ€˜s it!", "Thanks for testing :)");
    }
  }
};

const styles = StyleSheet.create({
  image: {
    marginBottom: 40,
    resizeMode: "contain",
    flex: 1,
    width: null
  },
  text: {
    ...TextStyles.largeText,
    textAlign: "center"
  },
  container: {
    flex: 1,
    padding: 30,
    paddingTop: 110,
    paddingBottom: 110,
    width: null,
    backgroundColor: "transparent",
    justifyContent: "center",
    alignItems: "stretch"
  },
  button: {
    backgroundColor: Colors.YELLOW,
    paddingVertical: 10,
    borderRadius: 20,
    width: 200,
    marginTop: 20
  }
});
