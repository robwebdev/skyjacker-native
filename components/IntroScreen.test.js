import "react-native";

import IntroScreen from "./IntroScreen";
import React from "react";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <IntroScreen
        image={require("../img/first-run/1_FirstRun_Radar.png")}
        titleText="Whatâ€˜s flying above you?"
        bodyText={["Check your Radar to", "see real airplanes nearby"]}
        buttonText="Okay..."
        nextScreen="Intro2"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
