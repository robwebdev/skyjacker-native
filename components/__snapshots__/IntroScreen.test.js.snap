// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  animation="fadeInRightBig"
  duration={500}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
      "padding": 30,
      "paddingBottom": 110,
      "paddingTop": 110,
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "flex": 1,
        "marginBottom": 40,
        "resizeMode": "contain",
        "width": null,
      }
    }
  />
  <View
    animation="pulse"
    delay={500}
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 20,
            "textAlign": "center",
          },
          Object {
            "color": "#FFCC00",
          },
        ]
      }
    >
      Whatâ€˜s flying above you?
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "lineHeight": 20,
          "textAlign": "center",
        }
      }
    >
      Check your Radar to
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "lineHeight": 20,
          "textAlign": "center",
        }
      }
    >
      see real airplanes nearby
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFCC00",
          "borderRadius": 20,
          "marginTop": 20,
          "opacity": 1,
          "paddingVertical": 10,
          "width": 200,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "lineHeight": 20,
              "textAlign": "center",
            },
            Object {
              "color": "black",
              "textAlign": "center",
            },
          ]
        }
      >
        Okay...
      </Text>
    </View>
  </View>
</View>
`;
